src/library_wasi.js

```diff
@@ -263,24 +263,159 @@ var WasiLibrary = {
     if (stream.getdents && offset === 0 && whence === {{{ cDefine('SEEK_SET') }}}) stream.getdents = null; // reset readdir state
     return 0;
   },
+
+#ifdef PURE_WASI
+  $preopens: "{3: '/'}",
```

Comment:
what is 3, and why is it `/`?

Reply:
Clearly this needs work yes :)

This basically says expose the whole of emscriptens VFS (the virtual filesytem created by the FS code) and make that available to WASI via the preopen system.   Pre-open file descriptors start at 3, right after stdin, stdout and stderr.   

This really needs integration with the FS layer... I think I should probably fix that even before landing this..